File: server/Models/Menu.js
const mongoose = require('mongoose')

const SubmenuSchema = mongoose.Schema({
    name:{
        type: String
    },
    path:{
        type: String
    }
})

const childrensSchema = mongoose.Schema({
    name:{
        type: String
    },
    icon:{
        type: String
    },
    patch:{
        type: String
    },
    submenu:[SubmenuSchema]
})

const MenuSchema = mongoose.Schema({

    name:{
        type: String
    },
    icon:{
        type: String
    },
    patch:{
        type: String
    },
    childrens:[childrensSchema]
})




module.exports = mongoose.model('menu',MenuSchema)

File: Go_api/models/menu.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Submenu struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"-"`
	Name string `bson:"name" json:"name"`
	Path string `bson:"path" json:"path"`
}

type Children struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"-"`
	Name    string    `bson:"name" json:"name"`
	Icon    string    `bson:"icon" json:"icon"`
	Patch   string    `bson:"patch" json:"patch"`
	Submenu []Submenu `bson:"submenu" json:"submenu"`
}

type Menu struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"-"`
	Name      string             `bson:"name" json:"name"`
	Icon      string             `bson:"icon" json:"icon"`
	Patch     string             `bson:"patch" json:"-"`
	Childrens []Children         `bson:"childrens" json:"childrens"`
}
